addon/-private/system/store.js

```diff
@@ -2065,32 +2047,53 @@ Store = Service.extend({
     @return {DS.Model}
   */
   modelFor(modelName) {
-    heimdall.increment(modelFor);
     assert("You need to pass a model name to the store's modelFor method", isPresent(modelName));
     assert('Passing classes to store methods has been removed. Please pass a dasherized string instead of '+ inspect(modelName), typeof modelName === 'string');
 
-    let factory = this.modelFactoryFor(modelName);
-    if (!factory) {
-      //Support looking up mixins as base types for polymorphic relationships
-      factory = this._modelForMixin(modelName);
-    }
+    let trueModelName = this._classKeyFor(modelName);
+
+    return this._modelFor(trueModelName);
+  },
+
+  /*
+    @private
+   */
+  _modelFor(modelName) {
```

Comment:
It remains public, the differentiator is that "public" APIs must always normalize the modelName while "private" APIs can assume a normalized modelName.  We currently call this so often internally I wanted to avoid that unnecessary step.